src/payment/strategies/cybersource/cybersource-payment-strategy.ts

```diff
@@ -0,0 +1,140 @@
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _cardinalClient: CardinalClient
+    ) {}
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId)).then( state => {
+            this._paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+            if (!this._paymentMethod || !this._paymentMethod.config || this._paymentMethod.config.testMode === undefined) {
+                throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+            }
+
+            return this._cardinalClient.initialize(this._paymentMethod.config.testMode).then(() => this._store.getState());
+        });
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment || !payment.paymentData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.config) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return !this._paymentMethod.config.is3dsEnabled || !payment.paymentData.ccNumber ? this._placeOrder(order, payment, payment.paymentData, options) :
+                this._placeOrderUsing3DS(order, payment, payment.paymentData as CreditCardInstrument, options, this._paymentMethod.clientToken);
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _placeOrderUsing3DS(order: OrderRequestBody, payment: OrderPaymentRequestBody, paymentData: CreditCardInstrument, options?: PaymentRequestOptions, clientToken?: string): Promise<InternalCheckoutSelectors> {
+        if (!clientToken) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPaymentMethod));
+        }
+
+        return this._cardinalClient.configure(clientToken).then(() => {
+            return this._cardinalClient.runBindProcess(paymentData.ccNumber).then(() => {
+                return this._placeOrder(order, payment, paymentData, options).catch(error => {
+                    if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'enrolled_card' })) {
+                        return Promise.reject(error);
+                    }
+
+                    return this._cardinalClient.getThreeDSecureData(error.body.three_ds_result, this._getOrderData(paymentData)).then(jwt =>
```

Comment:
If you do the change in `getThreeDSecureData` to return the whole object, then you can just change this:

```ts
const { three_ds_result: threeDResult } = error.body;

return this._cardinalClient.getThreeDSecureData(
		error.body.three_ds_result, 
		this._getOrderData(paymentData)
	)
    .then(threeDSecure => this.executePayment({
        ...payment,
        paymentData: {
            ...paymentData,
            threeDSecure,
        },
    }));
```